### PR TITLE
Rework of platformIO scripts in order to have a GUI to load examples easily.

### PR DESCRIPTION
**Pull request by jalinei**
**Major Features**

The new ergonics permits to load an extra platformIO library containing owntech examples. The library.json contained in the library is parsed by a script that create PlatformIO targets. 
Each target can be loaded from the GUI. 
This major leap forward was possible because of finding how to retrieve targets info before build using 

from SCons.Script import COMMAND_LINE_TARGETS # pylint: disable=import-error
Retrieving COMMAND_LINE_TARGETS permits to check which target will be built, and pre-build actions can be declared in a pre. script.
In this case, when an example is loaded, the whole user src/ directory is stored in a old/ directory, preventing accidental loss of important files. 
The library example is copied and pasted in src/ folder, and the build/ repertory is supressed to trigger a clean build operation.
A new build Middleware has been added, it removes owntech examples from the build system so that all examples are not compiled for nothing. 

**Applications**
Each applications can have it's own CONFIG calls by adding an app.conf file in src/directory. 
Similarly applications can have a dedicated app.overlay to tweak special feature using dts.
This is already leveraged with the example library in which some example require CONFIG_COMMUNICATION_API=y 
In that case, app.conf contains this config.

**Libraries**
Each library can have their own extra lib.conf file to handle pre-requised. For instance Control Library necessitate extra CMSIS Config that should not necessarily be added in root prj.conf. As such it made more sense to add a local per library lib.conf file. 
This required to pass an extra argument from platformio to cmake using :

board_build.zephyr.cmake_extra_args = -DBUILD_ENV_NAME=FOO

This flag is used in CMake to create the full path to the libraries in the correct platformIO environment. As such libdep can still be used in a per environment basis without risk of missing CONFIG calls, causing build failures.

**Minor features**
An extra GUI button has been added to switch between Normal and Expert Mode
Added an action to flash OwnTech bootloader using ST bootloader and dfu-util.
In STlink environment, console is automatically set to LPUART1 by changing 
zephyr,console = &cdc_acm_uart0; to 
zephyr,console = &lpuart1; 

**Naming revamp**
bootloader_serial is now USB
bootloader_stlink is now STlink
Targets that were previously in Platform are now located in OwnTech.

### CODE REVIEW

**Comment by jalinei** [comment]
**New scripts tested on Windows today - no specific issues found.**

Bootloader recovery requires to install WinUSB driver using Zadig software.
Note that this problem as its Linux counterpart: Installing udev rules for dfu-util. 
This complexity will require dedicated documentation. 

I encountered quite a few situation where the board was stucked and I had to reset, without being able to understand what was the issue.. It will need further testing to understand the root cause of the annoyance, but it is not a showstopper.

**Comment by jalinei** [comment]
@cfoucher-laas  ready to merge, please stash commits.

File: owntech/scripts/pre_target_examples.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        old_folder_count = 0
+        while os.path.exists(os.path.join(old_dir, f"old{old_folder_count}")):
+            old_folder_count += 1
+
+        current_old_dir = os.path.join(old_dir, f"old{old_folder_count}")
+        os.makedirs(current_old_dir)
+
+        if os.path.exists(src_dir):
+            shutil.move(src_dir, current_old_dir)
+        print(f"Previous main saved in {current_old_dir}")
```
Comment (rrodgeur) [inline owntech/scripts/pre_target_examples.py:58]: During my own tests I have not seen pass the message concerning the source files displacements. May be we have to search other ways to alert the user we are moving its files....no?

File: platformio.ini [hunk 2]
```diff
@@ -12,11 +12,14 @@
 # Available environments:
 # - bootloader_serial (default) -- Upload to the board using USB UART
```
Comment (rrodgeur) [inline platformio.ini:13]: comments to update ?

File: zephyr/CMakeLists.txt [hunk 3]
```diff
@@ -16,12 +16,55 @@ MACRO(SUBDIRLIST result curdir)
   SET(${result} ${dirlist})
 ENDMACRO()
 
+# This macro check for app.extension in curdir folder, where curdir and extension
+# are given as inputs. If found it returns the absolute path of app.extension.
+MACRO(FILELIST result curdir extension)
```
Comment (rrodgeur) [inline zephyr/CMakeLists.txt:21]: little remark: It should be possible to have only one macro...

**Pull request merged**